### PR TITLE
unixPB: add solaris x64 support

### PR DESCRIPTION
**Pull request by gdams**
The main pain with this is that the get_uri module doesn't work on Solaris so we have to use wget

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml:57]: We say "Not needed on Solaris" yet we still seem to have it in the common vars. Is this correct? I'd normally expect the ccache rol not to go through if it detects another version installed, so this guard seems irrelevant

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml:68]: Gut feel is still that these guards would be better inside the role, so that all the logic relating to the available versions is in one place, but I wouldn't block the PR on that basis at the moment.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:85]: Is cups not available through pkgutil?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:132]: Do we really need this for `sha256sum`? That would look rather odd

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:150]: What fails when this is not at the start of the PATH - it's generally preferable to have non-system-default tools at the end

(I've got a feeling there was a reason for this in the past but we need to document why in here since it's far from ideal)

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/main.yml:61]: These conditionals are looking a bit odd. The comment says "for Redhat" but there seems nothing to guard that. If it's linux-specific perhaps we should be explicit about that instead of saying "Not blah" when something new shows up in the playbooks

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:23]: Somewhat disappointing if this isn't the default

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml:108]: I take it this is because unarchive isn't working due to a cert error too?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml:27]: Why have we gone for `unarchive` here with `validate_certs: false` when we've used `wget` elsewhere?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/maven/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/maven/tasks/main.yml:24]: Same comment as freemarker ... why use a differnet method for these when using `wget` elsewhere?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/maven/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/maven/tasks/main.yml:27]: As per previous comments

File: ansible/plugins/inventory/adoptopenjdk_yaml.py
Comment (sxa) [inline ansible/plugins/inventory/adoptopenjdk_yaml.py:50]: Should we make this `packet-esxi` to make it clearer what the provider is? (Arguably doesn't need the `-esxi` but I would definitely be happier if it said packet somewhere)

**Review by sxa — CHANGES_REQUESTED** [review]
Sorry - quite a few comments :-) A few will just be questions but there are a few inconsistencies too

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/maven/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/maven/tasks/main.yml:24]: I've used get_url wherever possible. It seems to be down to the TLS version that each server accepts

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml:27]: I've used get_url wherever possible. It seems to be down to the TLS version that each server accepts

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml:56]: my understanding is that on nearly all Solaris systems the default is `e1000g0`. I'm not very happy with this approach for setting the hostname in general - it seems very hacky

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Solaris.yml:13]: Not technically needed but our other build machine used this jdk8 to connect to jenkins and the adopt one for buildJDK only. I was simply mirroring the setup

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml:57]: updated, the check only looks in `/usr/local/bin/ccache` which isn't where it's located on Solaris. I've updated the text.

File: ansible/plugins/inventory/adoptopenjdk_yaml.py
Comment (gdams) [reply ansible/plugins/inventory/adoptopenjdk_yaml.py:50]: updated

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:132]: yup it's a required symlink. As part of the build scripts we run sha256sum to generate the checkum

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:150]: OpenJDK expects all the gnutools to be present on Solaris. As part of the build scripts we currently `export PATH=/opt/csw/bin/:$PATH`

I was simply modifying the profile so that when people SSH in the environment is the same

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:23]: unfortunately, the built-in ansible user module doesn't create the home folder so this was a workaround

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml:68]: I'm currently working on the macOS playbook which will use this role. I'll make those changes over there

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/main.yml:61]: Agreed, I didn't want to break anything so the simplest solution was to skip Solaris. We should clarify if this is Rhel or Linux specific

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2017/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2017/tasks/main.yml:17]: Erm, this is nothing to do with Solaris so should not form part of this PR

Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2017/tasks/main.yml:17]: This is not related to Solaris so should not be in here

**Review by sxa — CHANGES_REQUESTED** [review]
Commit message should read `unixPB` at start for consistency instead of `ansible:` but this can be fixed on squashing. Otherwise happy to approve once the non-Solaris bit that's wormed it's way into this PR is removed ;-)

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2017/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2017/tasks/main.yml]: rebase sorted this out. looks like a git glitch

**Comment by gdams** [comment]
@sxa updated PTAL

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml:56]: Certainly on Solaris 10 I've had interface names of `bge0` although I'd suggest adding a comment to this saying that it may not work if the interface name is not `e1000g0`, or even `TODO: Allow other interfaces` above this section

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:132]: I know `sha256sum` is used but what tries to invoke it as `gsha256sum` which seems to be what this symlink operation is creating unless I'm missing something?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:150]: I really don't think that's a good idea. We can mask problems that users will potentially see if we change the OS default tools. We've already hit issues on AIX because of this sort of change and I don't want to risk testing being invalid as a result of the tools not being the standard solaris ones

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/freemarker/tasks/main.yml:27]: Hmmm is it TLS rather than the CA certs? (the latter should be replaceable)

**Comment by sxa** [comment]
Comments added. Let's try and sort this out tomorrow if possible.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:132]: Other way round, it symlinks gsha256sum to sha256sum. Also gmake to make etc

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:22]: `http` on here makes me very nervous although I appreciate that it's due to the same reason get_uri doesn't work.
What is `action=nocheck` required for as it seems to install ok without it?
I'd change it to this to explicitly install `CSWpkgutil` from the bundle
```suggestion
    yes | pkgadd -d http://get.opencsw.org/now CSWpkgutil
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:25]: What were these two lines for? Debugging? If we're downloading the latest in the initial `pkgadd` step this seems unnecessary

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:112]: Since the version we want isn't in the sfw/csw repos would it not be preferable to put this into `/usr/local/cups` instead of other namespaces?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:6]: I would strongly advise not setting this to `/bin/bash` as it's not guaranteed to be there on a basic Solaris installation

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:7]: Why are we ignoring errors here?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:132]: Eeeow - surprised it creates that one with a g prefix by default. I'm still not sure about the other two though - pretty sure we have stuff in our scripts that compensates for replacing tar and make with the g prefixed ones (search gtar and gmake in the build repo), and as per other comment I don't think it's right to switch the defaults on the system and potentially cause unecessary confusion

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Solaris.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Solaris.yml:13]: OK - I'd rather "use what we sell" but I won't consider it a blocker to this PR :-)

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:22]: Thinking some more we should possible store our own copy of this in Vendor_Files (or other comparable location) if we can't update the CA certs that pkgadd is using

Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:22]: Hmmm in fact given that the CSWpkgutil package is only about 220K gzipped I could just about accept having it stored in this repository and then copied acrosss to the machine to be installed (something I would normally NEVER advocate, but it's very small)

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:7]: This was copied from the Jenkins user step. The create user role fails on Solaris and macOS because of home directory issues but does actually create the user so the skip step works around this.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml:56]: updated

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:112]: This is the place that OpenJDK expects cups to be located. It's also where we put cups on the other solaris servers

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:7]: What's the issue on macos? We could set a variable for the home dir location and pass it into this operation with `home=/export/home/zeus` or wherever else it is (defaulting to `/home/zeus`. We run the risk of playbooks completing without anyone realising that it hasn't done the operations if we ignore errors here.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml]: updated PTAL

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml]: removed

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml]: updated

**Comment by gdams** [comment]
@sxa updated PTAL

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml:154]: @gdams Can you remove this part - I don't want that as a default on the machine in case it has other side effects or masks other problems. Happy to approve & merge this if you remove it.

**Comment by gdams** [comment]
@sxa updated PTAL

**Review by sxa — APPROVED** [review]
Thanks for the changes - happy for this to go in now but we need to keep an eye on the new workflows to ensure they don't cause any problems.

**Pull request merged**
